### DISCUSSION TITLE
Why `setErrors()` doesn't trigger child memo component render?

### DISCUSSION THREAD

**Discussion by Cow258** [react-hook-form/react-hook-form#10621] [Q&A] [answered]
Try the click `Set Error` and `Clear Error` button
Then the `MemoComp` will not be render expected

https://codesandbox.io/s/react-hook-form-ggrgsk?file=/src/App.js

![螢幕錄製 2023-07-01 21 49 33](https://github.com/react-hook-form/react-hook-form/assets/14261588/818445e7-641c-4de4-b152-a7411d123ad9)

![image](https://github.com/react-hook-form/react-hook-form/assets/14261588/ab539933-8d7e-48b7-9365-cb31bb0eccaa)

**Comment by leapful** [comment, marked as answer]
@Cow258 

Object `errors` will not change its reference via `setErrors()` method as it's a root error object and will be accessed via `formState` proxy.

If you want to get `errors` from memorized component, just use `useFormState` instead.

Working demo: https://codesandbox.io/s/react-hook-form-forked-qz6rkg?file=/src/App.js
